### ISSUE TITLE
[FEATURE REQUEST] Using the tag system in `devicon.json`

### ISSUE THREAD

**Issue by Thomas-Boi** [devicons/devicon#988]
### Problem

Each entry in the `devicon.json` has a `tag` attribute. As far as I know, this is never used both in our official website or during our icon creation pipeline. Since we have it, we should probably use it.

### Suggested feature

I can think of two choices:
1. Create a search system using the tags for our official website. Users can specify a tag and we will filter the result for them based on that tag. They can also specify multiple tags and we will only search result that satisfy all the tags (`AND` relationship).
2. Remove the `tag` attribute. If we don't use it, might as well remove it. Makes writing the entry easier for contributors and shorten the json file.

### Additional information

I'm leaning towards option 1 just because I think it's cool to have that functionality. However, we then run into the following issues:

- Standardizing the tags: perhaps we have a `tags.json` or a `Tags` page in our wiki that explains the tagging. We can then validate them using `check-bot`.
- Converting the old tags to the new standardized tags. This has to be done manually and we have around 200 entries to check.
- Coding it to the site. I'm not sure how easy it would be to use AngularJS to create this feature. I'm confident it's feasible since I can just look at the search code and change it a bit. However, this is a bit of a wild variable.

**Comment by Snailedlt** [comment]
I'm also leaning towards option 1 here. Adding some sort of categorization on the website would improve it a lot imo. It's not really needed, but it's still nice to have :)

Restricting what tags are allowed and which are not is imo a must though. It's ridiculous that we don't have this in place already, but I guess it comes down to effort being put elsewhere which I guess is okey when the tags aren't used anywhere (yet) anyways. I think we could pretty easily add a list of words to the `check-bot`, but not sure if that's the best solution, since people won't get any validation if their tag is valid or not until they make a PR or run the `check-bot` themselves. It might be better to have a [JSON Schema](https://json-schema.org/) with the [valid values in an enum](https://stackoverflow.com/a/38049340/12206312), that way you get some free Intellisense so that the contributors can catch errors before making a PR.

**Comment by Snailedlt** [comment]
@BenSouchet @Panquesito7 You guys have any thoughts on this?

**Comment by BenSouchet** [comment]
I agree to fact that it would be good to use the tags and unify them to avoid having too much tags, don't know how to properly ensure that new addition use valids tags, don't know http://json-schema.org/ but seems to be a good option.

Adding a list of tags would maybe "poluate" the devicon website homepage, don't know how to display tags without adding too much elements in the interface. I'm incline as a first step to allow user search tags in the searchbar, for exemple typing "editor" will display all icons that has the tag "editor", currently we get no results :
![currently](https://user-images.githubusercontent.com/17025808/193533245-db0499a5-e819-4075-8fc5-266cf593ac74.png)

I think the tags isn't currently used because no one had the time or the inclination to implement these features, this work can be time and energy consuming :)

@Snailedlt If you and / or someone else want to dive into this feature do not hesitate, it's indeed a true key missing feature, that could help lots of people.

**Comment by Snailedlt** [comment]
I think if we add tags to the website they should be separate from the searchfield. Maybe just a `filter` dropdown next to it?

I also haven't used json-schema before, but it seems like the way to go.

Lastly I wanna work on this, but I'm not sure I have time to do so any time soon. I spend most of my time with devicon doing maintainer stuff like reviews, labeling and closing issues. There's also a few devops issues I want to get done before this one... But if I get time after that, I'll start implementing this too. In the meantime anyone should feel free to try their hand at this issue :)

**Comment by Panquesito7** [comment]
I go for option 1. Sounds like a much better idea.
We can document all the possible valid tags and add them to a list. If someone wants to add a new tag, they can create a PR.

**Comment by Snailedlt** [comment]
Sounds like something to look at if you're reworking the website @lunatic-fox :)
